src/Dataset.ts

```diff
@@ -693,21 +693,27 @@ export class JobUpload {
       fileSize = fileOrPath.size;
       fileName = fileOrPath.name;
     }
+    // we want to try to get a sticky session cookie for TUS uploads
+    const headers: { [name: string]: string } = { Authorization: "Bearer " + this._config.app["_config"].token };
+    await setStickySessionCookie(headers, this._config.app["_config"].url);
+
+    let previousChunkCompleted = Date.now();
     return new Promise<void>((resolve, reject) => {
       const upload = new tus.Upload(rs, {
         endpoint: this.jobUrl + "/add",
         metadata: {
           filename: fileName,
         },
-        headers: {
-          Authorization: "Bearer " + this._config.app["_config"].token,
-        },
+        headers,
         mapUrl: this.urlMapper,
         chunkSize: 5 * 1024 * 1024,
         uploadSize: fileSize,
         retryDelays: [2000, 5000, 10000, 40000, 50000],
         onError: reject,
-        onProgress: function (_bytesUploaded: number, _bytesTotal: number) {},
+        onChunkComplete: function (...stats) {
+          log(formatUploadProgress(Date.now() - previousChunkCompleted, ...stats));
```

Comment:
:thinking: For really large uploads this might flood the logs in pipelines. I think given the data you could also determine when the rounded percentage increases and only then log the progress. Do you think that is worth implementing?

Reply:
These logs are only printed when someone sets the `DEBUG` environment variable, which I don't think is ever set in (production) ETLs (I'm assuming you mean ETL pipelines).

Reply:
See [this line](https://github.com/TriplyDB/TriplyDB-JS/pull/28/files/0340773370fcf0044b83e6e883a61df01340164e#diff-2126a2a0d3a8e8200f8d830a3342c4aadaca2a50f227e5f6a6f1c595cabd865eR7)